pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -93,23 +92,31 @@ func doUpdateContentTree(ctx *volumemgrContext, status *types.ContentTreeStatus)
 			}
 
 			// at this point, we will have the hash of the root blob as status.ContentSha256,
-			// so we need to create the BlobStatus, if it does not exist
-			// if it does not already exist
-			if lookupOrCreateBlobStatus(ctx, sv, status.ContentSha256) == nil {
-				rootBlob := &types.BlobStatus{
+			// so we need to create the BlobStatus, if it does not exist already
+			rootBlob := lookupOrCreateBlobStatus(ctx, sv, status.ContentSha256)
+			if rootBlob == nil {
+				rootBlob = &types.BlobStatus{
 					DatastoreID: status.DatastoreID,
 					RelativeURL: status.RelativeURL,
 					Sha256:      status.ContentSha256,
 					Size:        status.MaxDownloadSize,
 					State:       types.INITIAL,
 					BlobType:    types.BlobUnknown,
 				}
-				log.Infof("doUpdateContentTree: publishing root BlobStatus (%s) for content tree (%s)",
+				log.Infof("doUpdateContentTree: publishing new root BlobStatus (%s) for content tree (%s)",
+					status.ContentSha256, status.ContentID)
+				publishBlobStatus(ctx, rootBlob)
+			} else if rootBlob.IsLoaded {
+				//Need to update DatastoreID and RelativeURL if the blob is already loaded into CAS,
+				// because if any child blob is not downloaded, then we would need the below data.
+				rootBlob.DatastoreID = status.DatastoreID
+				rootBlob.RelativeURL = status.RelativeURL
```

Comment:
In doing this, are we not changing the `BlobStatus` itself? Is the `DatastoreID` of the `BlobStatus` _actually_ correct? What if we got the original blob from one place, but the next one will come from another? What if we have 2 in parallel that have different datastores but the same content (it can and does happen, often)? This will mess it up.

We should not be modifying the `BlobStatus` to add the `DatastoreID` and `RelativeURL`, but rather passing it on when creating ` DownloaderConfig`.

Reply:
FYI, this one is OK to leave for now, and we can pull a separate smaller PR afterwards to fix it, if it is too complex.